src/ppx/grfx/vk/vk_device.cpp

```diff
@@ -297,6 +297,63 @@ Result Device::ConfigureFeatures(const grfx::DeviceCreateInfo* pCreateInfo, VkPh
+    diFeatures.shaderStorageTexelBufferArrayDynamicIndexing       = foundDiFeatures.shaderStorageTexelBufferArrayDynamicIndexing;
+    diFeatures.shaderUniformBufferArrayNonUniformIndexing         = foundDiFeatures.shaderUniformBufferArrayNonUniformIndexing;
+    diFeatures.shaderSampledImageArrayNonUniformIndexing          = foundDiFeatures.shaderSampledImageArrayNonUniformIndexing;
+    diFeatures.shaderStorageBufferArrayNonUniformIndexing         = foundDiFeatures.shaderStorageBufferArrayNonUniformIndexing;
+    diFeatures.shaderStorageImageArrayNonUniformIndexing          = foundDiFeatures.shaderStorageImageArrayNonUniformIndexing;
+    diFeatures.shaderInputAttachmentArrayNonUniformIndexing       = foundDiFeatures.shaderInputAttachmentArrayNonUniformIndexing;
+    diFeatures.shaderUniformTexelBufferArrayNonUniformIndexing    = foundDiFeatures.shaderUniformTexelBufferArrayNonUniformIndexing;
+    diFeatures.shaderStorageTexelBufferArrayNonUniformIndexing    = foundDiFeatures.shaderStorageTexelBufferArrayNonUniformIndexing;
+    diFeatures.descriptorBindingUniformBufferUpdateAfterBind      = foundDiFeatures.descriptorBindingUniformBufferUpdateAfterBind;
+    diFeatures.descriptorBindingSampledImageUpdateAfterBind       = foundDiFeatures.descriptorBindingSampledImageUpdateAfterBind;
+    diFeatures.descriptorBindingStorageImageUpdateAfterBind       = foundDiFeatures.descriptorBindingStorageImageUpdateAfterBind;
+    diFeatures.descriptorBindingStorageBufferUpdateAfterBind      = foundDiFeatures.descriptorBindingStorageBufferUpdateAfterBind;
+    diFeatures.descriptorBindingUniformTexelBufferUpdateAfterBind = foundDiFeatures.descriptorBindingUniformTexelBufferUpdateAfterBind;
+    diFeatures.descriptorBindingStorageTexelBufferUpdateAfterBind = foundDiFeatures.descriptorBindingStorageTexelBufferUpdateAfterBind;
+    diFeatures.descriptorBindingUpdateUnusedWhilePending          = foundDiFeatures.descriptorBindingUpdateUnusedWhilePending;
+    diFeatures.descriptorBindingPartiallyBound                    = foundDiFeatures.descriptorBindingPartiallyBound;
+    diFeatures.descriptorBindingVariableDescriptorCount           = foundDiFeatures.descriptorBindingVariableDescriptorCount;
+    diFeatures.runtimeDescriptorArray                             = VK_TRUE;
+
+    // Verify that any asserted features were actually found to be
+    // supported.
+    std::vector<std::string_view> missingFeatures;
+    if (!foundDiFeatures.runtimeDescriptorArray) {
+        missingFeatures.push_back("runtimeDescriptorArray");
+    }
+
+    if (!missingFeatures.empty()) {
+        std::stringstream ss;
+        ss << "Device does not support required features:" << PPX_LOG_ENDL;
+        for (const auto& elem : missingFeatures) {
+            ss << " " << elem << PPX_LOG_ENDL;
+        }
+        PPX_ASSERT_MSG(false, ss.str());
+        return ppx::ERROR_REQUIRED_FEATURE_UNAVAILABLE;
+    }
+
```

Comment:
Why is this not:
```cpp
PPX_ASSERT_MSG(foundDiFeatures.runtimeDescriptorArray, "RuntimeDescriptorArray feature missing.");
return foundDiFeatures.runtimeDescriptorArray ? PPX::SUCCESS : PPX::ERROR_REQUIRED_FEATURE_UNAVAILABLE;
```

Reply:
I wanted to keep it like this so that we can add more checks in the future